CommonTools/TriggerUtils/src/GenericTriggerEventFlag.cc

```diff
@@ -21,9 +21,19 @@ static const bool useL1EventSetup( true );
 static const bool useL1GtTriggerMenuLite( false );
 
 
+GenericTriggerEventFlag::GenericTriggerEventFlag( const edm::ParameterSet & config, edm::ConsumesCollector & iC):
+  GenericTriggerEventFlag(config,iC,false)
+{
+  if ( config.exists( "andOrL1" ) ) {
```

Comment:
Hi David, 

Many thanks for the review. So I think this not strictly needed but it does simplify things for me. 

I wanted to leave the original public constructor: 
https://github.com/Sam-Harper/cmssw/blob/9636bcf9626f001993f6d941c1b598a8b7eb3a13/CommonTools/TriggerUtils/interface/GenericTriggerEventFlag.h#L157-L169

exactly as is. Which means all the  l1 setup gets done outside the private constructor (which is now tagged with an extra argument of a bool). A dirty little secret, that bool is only there to identify the private contructor, 

https://github.com/Sam-Harper/cmssw/blob/9636bcf9626f001993f6d941c1b598a8b7eb3a13/CommonTools/TriggerUtils/src/GenericTriggerEventFlag.cc#L130-L135

could easily just be at the location you mention and inside the config.exists("andOrL1") statement.  But I put it in the private constructor simply because I didnt like having an unused variable. 

As written, it is simpler to have all the L1 construction happening in the public constructors and a shared private constructor which does the rest of the setup. Note, the false only means it is an error that stage-1 is selected, stage-2 can be setup.